pkg/distributors/test/MerkleRedeem.test.ts

```diff
@@ -0,0 +1,273 @@
+      const merkleProof = merkleTree.getHexProof(elements[0]);
+
+      await merkleRedeem.connect(lp2).claimWeek(lp2.address, 1, claimedBalance, merkleProof, false);
+
+      const errorMsg = 'cannot claim twice';
+      expect(
+        merkleRedeem.connect(lp2).claimWeek(lp2.address, 1, claimedBalance, merkleProof, false)
+      ).to.be.revertedWith(errorMsg);
+    });
+  });
+
+  describe('When a user has several allocation to claim', () => {
+    const claimBalance1 = bn('1000');
+    const claimBalance2 = bn('1234');
+
+    let elements1: string[];
+    let merkleTree1: MerkleTree;
+    let root1: string;
+
+    let elements2: string[];
+    let merkleTree2: MerkleTree;
+    let root2: string;
+
+    beforeEach(async () => {
+      elements1 = [encodeElement(lp2.address, claimBalance1)];
+      merkleTree1 = new MerkleTree(elements1);
+      root1 = merkleTree1.getHexRoot();
+
+      elements2 = [encodeElement(lp2.address, claimBalance2)];
+      merkleTree2 = new MerkleTree(elements2);
+      root2 = merkleTree2.getHexRoot();
+
+      await merkleRedeem.seedAllocations(bn(1), root1, claimBalance1);
+
+      await merkleRedeem.seedAllocations(bn(2), root2, claimBalance2);
+    });
+
+    it('Allows the user to claim multiple weeks at once', async () => {
+      const claimedBalance1 = bn('1000');
+      const claimedBalance2 = bn('1234');
+
+      const proof1: BytesLike[] = merkleTree1.getHexProof(elements1[0]);
+      const proof2: BytesLike[] = merkleTree2.getHexProof(elements2[0]);
+
+      const merkleProofs = [
+        { week: bn(1), balance: claimedBalance1, merkleProof: proof1 },
+        { week: bn(2), balance: claimedBalance2, merkleProof: proof2 },
+      ];
+
+      await merkleRedeem.connect(lp2).claimWeeks(lp2.address, merkleProofs, false);
+
+      const result = await rewardToken.balanceOf(lp2.address);
+      expect(result).to.equal(bn('2234')); //"user should receive all tokens, including current week"
+    });
+
+    it('Returns an array of week claims', async () => {
+      let expectedResult = [false, false];
+      let result = await merkleRedeem.claimStatus(lp2.address, 1, 2);
+      expect(result).to.eql(expectedResult); // "claim status should be accurate"
+      const claimedBalance1 = bn('1000');
+      const proof1 = merkleTree1.getHexProof(elements1[0]);
+
+      const merkleProofs = [{ week: bn(1), balance: claimedBalance1, merkleProof: proof1 }];
+
+      await merkleRedeem.connect(lp2).claimWeeks(lp2.address, merkleProofs, false);
+
+      expectedResult = [true, false];
+      result = await merkleRedeem.claimStatus(lp2.address, 1, 2);
+      expect(result).to.eql(expectedResult); // "claim status should be accurate"
+    });
+
+    it('Returns an array of merkle roots', async () => {
+      const expectedResult = [root1, root2];
+      const result = await merkleRedeem.merkleRoots(1, 2);
+      expect(result).to.eql(expectedResult); // "claim status should be accurate"
```

Comment:
Note that it is not the claim status we're testing here. Overall, I'd try to have these tests written in a way such that these comments are not needed because what they're testing is already clear from the test description. 

In this case, I don't think we need the comment at all. In other tests that test multiple things (e.g. `verifyClaim` works, `redeem` works, tokens are transferred, events are emitted, the claim status is updated), I'd consider splitting those in smaller tests that check each individual property.